### ISSUE TITLE
read_excel has one parameter -  mangle_dupe_cols to handle duplicate columns but fails if the duplicate columns are case sensitive.

### ISSUE THREAD

**Issue by saikrishnapujari102087** [databricks/koalas#2213]
mangle_dupe_cols - default is True
So ideally it should have handled duplicate columns, but in case if the columns are case sensitive it fails as below.

AnalysisException: Reference '`Sheet.col1`' is ambiguous, could be: Sheet.col1, Sheet.col1.

Where two columns are Col1 and cOl1

In the best practices, there is a mention of not to use case sensitive columns - https://koalas.readthedocs.io/en/latest/user_guide/best_practices.html#do-not-use-duplicated-column-names

Either the docs for read_excel/mangle_dupe_cols has to be updated about this or it has to be handled.

**Comment by itholic** [comment]
As commented in offline channel,

I think mangle_dupe_cols is not for handling the case sensitive, but it handles the case when the column names are exactly the same as below:

```python
>>> pdf  # Column names are exactly the same, and `mangle_dupe_cols` handle this case.
   A  A
0  1  4
1  2  5
2  3  6
>>> pdf.to_excel("dup.xlsx")
>>> pd.read_excel("dup.xlsx", index_col=0)
   A  A.1
0  1    4
1  2    5
2  3    6
>>> ks.read_excel("dup.xlsx", index_col=0)  # In this case, Koalas works the same as pandas
   A  A.1
0  1    4
1  2    5
2  3    6
```

I think your case is more similar to the example below:

```python
>>> pdf  # Column names are not the same, and `mangle_dupe_cols` is not for this case.
   A  a
0  1  4
1  2  5
2  3  6
>>> pdf.to_excel("dup.xlsx")
>>> pd.read_excel("dup.xlsx", index_col=0)
   A  a
0  1  4
1  2  5
2  3  6
>>> ks.read_excel("dup.xlsx", index_col=0)
Traceback (most recent call last):
...
pyspark.sql.utils.AnalysisException: Reference 'A' is ambiguous, could be: A, A.
```

In this case, you should spark.conf.set("spark.sql.caseSensitive","true") before reading the excel file like you did

**Comment by itholic** [comment]
Yeah, we should address this.

Would you mind file the issue to the [Apache Spark JIRA](https://issues.apache.org/jira/projects/SPARK/issues) ??

This repository is in maintenance mode, as Koalas has been moved in to the PySpark (pandas API on Spark).

**Comment by saikrishnapujari102087** [comment]
@itholic Yes, created - https://issues.apache.org/jira/browse/SPARK-38004
